### ISSUE TITLE
Add a option to freeze all app by one click and unfreeze all by one click

### ISSUE THREAD

**Issue by ghost** [oasisfeng/island#341]
Add a option to freeze all app by one click and unfreeze all by one click

**Comment by oasisfeng** [comment]
Such "Freeze all" is potentially dangerous, because which apps should be included in "all" is hard to decide. If some critical apps are frozen, your system may not work properly, or even refuse to work.

I'd suggest to try "Ice Box", which has a similar feature to unfreeze all apps added to it, in one click. Ice Box works seamlessly within Island.
https://play.google.com/store/apps/details?id=com.catchingnow.icebox

**Comment by ghost** [comment]
But It requires premium version for adding more than 5apps in list while island is completely free

**Comment by RARE-BiT** [comment]
This can be done by combining Island with Oasis Feng's other app Greenify. Just select the apps in Island and specify "auto-freeze with Greenify". In the Greenify app, you may need to additionally select the option "Ignore background-free" for the the app to force it to hibernate even though the app advertises no background use. When you hibernate from Greenify your apps will all deep-hibernate together. By default, apps that are actively being used, such as YouTube Music streaming music, will not be hibernated. If you choose the option "Ignore working state" it will hibernate the app even if it is in use. 

Another method for freezing apps in a single click is by disabling the work profile, but this is less selective than the Greenify method as it obviously freezes all apps in the Island.

**Comment by proletarius101** [comment]
The ability to freeze individual unfrozen apps right in Island is already there in the notification panel. But it's easy to open many usually frozen apps and flood the notification center. It would be nice to have a button to freeze them all in one click right in Island.

They are not necessarily in the Work Profile due to Managed Mainland. Nor do all apps in the work profile needs to be frozen.

**Comment by RARE-BiT** [comment]
It would be nice if Island had a freeze all option, but as I stated previously you can achieve the functionality you want with Greenify. In some ways it is a better option because it can be a "frequently re-frozen" subset list of apps that is specified in Greenify rather than freezing all, or you can choose all if desired.

I have the Greenify hibernate function mapped to a swipe-down gesture on my Tools & Settings app group. In the attached video, I swipe down to trigger the hibernate and then pull up the app drawer. You can see that every app from AntennaPod to YouTube Music is frozen and disappears from the app drawer one by one.

https://user-images.githubusercontent.com/49774464/184517038-afdca09f-d00d-4ded-a4c1-1bba4932cf3f.mp4

**Comment by proletarius101** [comment]
@Veazer Greenify is abandoned

**Comment by RARE-BiT** [comment]
True, and I can understand not wanting to pay for an app that hasn't had updates for 3 years and probably won't be updated. In terms of stability and functionality though, it is still working great and can accomplish precisely the thing you're trying to do. 

Maybe @oasisfeng can consider dropping the price of Greenify for Island users due to its stale status.

**Comment by RARE-BiT** [comment]
@proletarius101 - Have you looked at using [Ice Box](https://play.google.com/store/apps/details?id=com.catchingnow.icebox) as a trigger to freeze all apps? I haven't used it but it is suggested in the 'Discovery' tab of Island.

**Comment by Hydra129** [comment]
@proletarius101 Uh, your video is not working:

Https://www.mediafire.com/file/rmwxm1jnm7725kp/Vidma_Screen_Recorder_Fail_To_Play_Video_Scnsht_20220830-221645.png/file

👆 Throws up this error, after I downloaded it, & won't play in my browser [Chrome] either.

**Comment by RARE-BiT** [comment]
@Hydra129 That was my video, not proletarious101. 

Not sure why you can't play it, it works fine in VLC and Chrome for me and on the embedded  player used here on the Github post page. Like many screen capture videos, it has variable frame rate which some older players don't know how to handle so you might want to try something else.
